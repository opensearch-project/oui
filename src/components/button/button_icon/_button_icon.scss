/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

// This file has lots of modifiers and is somewhat nesty by nature
// sass-lint:disable nesting-depth

.ouiButtonIcon {
  @include ouiButton;
  @include ouiSlightShadow;

  border-radius: $ouiButtonBorderRadius;
  width: $ouiButtonHeight;

  // Ensures center alignment of any sized icon inside buttons and anchors
  display: inline-flex;
  align-items: center;
  justify-content: space-around;

  & > svg {
    // prevents the element and its children from receiving any pointer events to fix not bubbling click event in Safari
    // https://stackoverflow.com/questions/24078524/svg-click-events-not-firing-bubbling-when-using-use-element
    pointer-events: none;
  }

  &.ouiButtonIcon--empty {
    box-shadow: none !important; // sass-lint:disable-line no-important
    border: none;
  }

  &.ouiButtonIcon-isDisabled {
    @include ouiButtonContentDisabled;
    color: $ouiButtonColorDisabledText;
    border-color: $ouiButtonColorDisabled;

    &.ouiButtonIcon--fill {
      // Only increase the contrast of background color to text to 2.0 for disabled
      color: makeHighContrastColor($ouiButtonColorDisabled, $ouiButtonColorDisabled, 2);
      background-color: $ouiButtonColorDisabled;
      border-color: $ouiButtonColorDisabled;

      &:hover,
      &:focus,
      &:focus-within {
        background-color: $ouiButtonColorDisabled;
        border-color: $ouiButtonColorDisabled;
      }
    }

    &:hover,
    &:focus,
    &:focus-within {
      @include ouiSlightShadow;
      text-decoration: none;
    }
  }
}

.ouiButtonIcon--xSmall {
  height: $ouiButtonHeightXSmall;
  width: $ouiButtonHeightXSmall;
}

.ouiButtonIcon--small {
  height: $ouiButtonHeightSmall;
  width: $ouiButtonHeightSmall;
}

// Create button modifiers based upon the map.
@each $name, $color in $ouiButtonTypes {
  .ouiButtonIcon--#{$name} {
    @if ($name == 'ghost') {
      // Ghost is unique and ALWAYS sits against a dark background.
      color: $color;
    } @else if ($name == 'text') {
      // The default color is lighter than the normal text color, make the it the text color
      color: $ouiTextColor;
    } @else {
      // Other colors need to check their contrast against the page background color.
      color: makeHighContrastColor($color, $ouiPageBackgroundColor);
    }

    border-color: $color;

    &.ouiButtonIcon--fill {
      background-color: $color;
      border-color: $color;

      // The function makes that hexes safe for theming
      $fillTextColor: chooseLightOrDarkText($color, $ouiColorGhost, $ouiColorInk);

      color: $fillTextColor;

      &:not([class*='isDisabled']) {
        &:hover,
        &:focus,
        &:focus-within {
          background-color: darken($color, 5%);
          border-color: darken($color, 5%);
        }

        &:active {
          @include ouiSlightShadowActive(darken($color, 10%), .4);
        }
      }
    }

    &:not([class*='isDisabled']) {
      $shadowColor: $ouiShadowColor;
      @if ($name == 'ghost') {
        $shadowColor: $ouiColorInk;
      } @else if (lightness($ouiTextColor) < 50) {
        // Only if this is the light theme do we use the button variant color to colorize the shadow
        $shadowColor: desaturate($color, 60%);
      }

      @include ouiSlightShadow($shadowColor);

      &:hover,
      &:focus,
      &:focus-within {
        @include ouiSlightShadowHover($shadowColor);
        background-color: transparentize($color, .9);
      }

      &:active {
        @include ouiSlightShadowActive($color);
      }
    }
  }
}

// Fix ghost/disabled look specifically
.ouiButtonIcon.ouiButtonIcon-isDisabled.ouiButtonIcon--ghost {
  &,
  &:hover,
  &:focus,
  &:focus-within {
    @include ouiSlightShadow($ouiColorInk);
    color: $ouiButtonColorGhostDisabled;
    border-color: $ouiButtonColorGhostDisabled;
  }

  &.ouiButton--fill {
    background-color: $ouiButtonColorGhostDisabled;
    color: makeHighContrastColor($ouiButtonColorGhostDisabled, $ouiButtonColorGhostDisabled, 2);
  }
}
