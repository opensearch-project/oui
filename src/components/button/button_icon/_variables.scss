/*!
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

// Modifier naming and colors.
$ouiButtonIconTypes: (
  accent: $ouiColorAccentText,
  danger: $ouiColorDangerText,
  ghost: $ouiColorGhost,
  primary: $ouiColorPrimaryText,
  subdued: $ouiTextSubduedColor,
  success: $ouiColorSuccessText,
  text: $ouiColorDarkShade,
  warning: $ouiColorWarningText,
);


/* OUI -> EUI Aliases */
$euiButtonIconTypes: $ouiButtonIconTypes;

/* End of Aliases */
