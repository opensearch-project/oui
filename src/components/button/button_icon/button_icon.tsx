/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, {
  AnchorHTMLAttributes,
  ButtonHTMLAttributes,
  FunctionComponent,
  Ref,
} from 'react';
import classNames from 'classnames';

import { getSecureRelForTarget } from '../../../services';
import {
  CommonProps,
  ExclusiveUnion,
  PropsForAnchor,
  PropsForButton,
  keysOf,
} from '../../common';

import { IconType, IconSize, OuiIcon } from '../../icon';

import { validateHref } from '../../../services/security/href_validator';

export type OuiButtonIconColor =
  | 'accent'
  | 'danger'
  | 'ghost'
  | 'primary'
  /**
   * Set for deprecation 3/2/21
   * This color button is close enough to text to be duplicative
   */
  | 'subdued'
  | 'success'
  | 'text'
  | 'warning';

const displayToClassNameMap = {
  base: null,
  empty: 'ouiButtonIcon--empty',
  fill: 'ouiButtonIcon--fill',
};

export const DISPLAYS = keysOf(displayToClassNameMap);
type OuiButtonIconDisplay = keyof typeof displayToClassNameMap;

export interface OuiButtonIconProps extends CommonProps {
  iconType: IconType;
  /**
   * Any of the named color palette options.
   * **`subdued` set to be DEPRECATED, use `text` instead**
   */
  color?: OuiButtonIconColor;
  'aria-label'?: string;
  'aria-labelledby'?: string;
  isDisabled?: boolean;
  /**
   * Overall size of button.
   * Matches the sizes of other OuiButtons
   */
  size?: OuiButtonIconSizes;
  /**
   * Size of the icon only.
   * This will not affect the overall size of the button
   */
  iconSize?: IconSize;
  /**
   * Applies the boolean state as the `aria-pressed` property to create a toggle button.
   * *Only use when the readable text does not change between states.*
   */
  isSelected?: boolean;
  /**
   * Sets the display style for matching other OuiButton types.
   * `base` is equivelant to a typical OuiButton
   * `fill` is equivelant to a filled OuiButton
   * `empty` (default) is equivelant to an OuiButtonEmpty
   */
  display?: OuiButtonIconDisplay;
}

type OuiButtonIconPropsForAnchor = {
  type?: string;
} & PropsForAnchor<
  OuiButtonIconProps,
  {
    buttonRef?: Ref<HTMLAnchorElement>;
  }
>;

export type OuiButtonIconPropsForButton = {
  type?: 'submit' | 'reset' | 'button';
} & PropsForButton<
  OuiButtonIconProps,
  {
    buttonRef?: Ref<HTMLButtonElement>;
  }
>;

type Props = ExclusiveUnion<
  OuiButtonIconPropsForAnchor,
  OuiButtonIconPropsForButton
>;

const colorToClassNameMap: { [color in OuiButtonIconColor]: string } = {
  accent: 'ouiButtonIcon--accent',
  danger: 'ouiButtonIcon--danger',
  ghost: 'ouiButtonIcon--ghost',
  primary: 'ouiButtonIcon--primary',
  subdued: 'ouiButtonIcon--subdued',
  success: 'ouiButtonIcon--success',
  text: 'ouiButtonIcon--text',
  warning: 'ouiButtonIcon--warning',
};

export const COLORS = keysOf(colorToClassNameMap);

const sizeToClassNameMap = {
  xs: 'ouiButtonIcon--xSmall',
  s: 'ouiButtonIcon--small',
  m: 'ouiButtonIcon--medium',
};

export type OuiButtonIconSizes = keyof typeof sizeToClassNameMap;

export const SIZES = keysOf(sizeToClassNameMap);

export const OuiButtonIcon: FunctionComponent<Props> = ({
  className,
  iconType,
  iconSize = 'm',
  color = 'primary',
  isDisabled: _isDisabled,
  href,
  type = 'button',
  display = 'empty',
  target,
  rel,
  size = 'xs',
  buttonRef,
  isSelected,
  ...rest
}) => {
  const isHrefValid = !href || validateHref(href);
  const isDisabled = _isDisabled || !isHrefValid;

  const ariaHidden = rest['aria-hidden'];
  const isAriaHidden = ariaHidden === 'true' || ariaHidden === true;

  if (!rest['aria-label'] && !rest['aria-labelledby'] && !isAriaHidden) {
    console.warn(
      `OuiButtonIcon requires aria-label or aria-labelledby to be specified because icon-only
      buttons are screen-reader-inaccessible without them.`
    );
  }
  const classes = classNames(
    'ouiButtonIcon',
    {
      'ouiButtonIcon-isDisabled': isDisabled,
    },
    colorToClassNameMap[color],
    display && displayToClassNameMap[display],
    size && sizeToClassNameMap[size],
    className
  );

  // Add an icon to the button if one exists.
  let buttonIcon;

  if (iconType) {
    buttonIcon = (
      <OuiIcon
        className="ouiButtonIcon__icon"
        type={iconType}
        size={iconSize}
        aria-hidden="true"
        color="inherit" // forces the icon to inherit its parent color
      />
    );
  }

  // <a> elements don't respect the `disabled` attribute. So if we're disabled, we'll just pretend
  // this is a button and piggyback off its disabled styles.
  if (href && !isDisabled) {
    const secureRel = getSecureRelForTarget({ href, target, rel });

    return (
      <a
        tabIndex={isAriaHidden ? -1 : undefined}
        className={classes}
        href={href}
        target={target}
        rel={secureRel}
        ref={buttonRef as Ref<HTMLAnchorElement>}
        {...(rest as AnchorHTMLAttributes<HTMLAnchorElement>)}>
        {buttonIcon}
      </a>
    );
  }

  let buttonType: ButtonHTMLAttributes<HTMLButtonElement>['type'];
  return (
    <button
      tabIndex={isAriaHidden ? -1 : undefined}
      disabled={isDisabled}
      className={classes}
      aria-pressed={isSelected}
      type={type as typeof buttonType}
      ref={buttonRef as Ref<HTMLButtonElement>}
      {...(rest as ButtonHTMLAttributes<HTMLButtonElement>)}>
      {buttonIcon}
    </button>
  );
};

// @internal
export type OuiSmallButtonIconProps = Omit<OuiButtonIconProps, 'size'>;

// @internal
type SmallProps = ExclusiveUnion<
  Omit<OuiButtonIconPropsForAnchor, 'size'>,
  Omit<OuiButtonIconPropsForButton, 'size'>
>;

// @internal
export const OuiSmallButtonIcon: FunctionComponent<SmallProps> = (props) => (
  <OuiButtonIcon {...props} size="s" />
);
