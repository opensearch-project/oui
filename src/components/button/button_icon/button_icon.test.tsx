/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render, screen, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { requiredProps } from '../../../test/required_props';

import { OuiButtonIcon, COLORS, DISPLAYS, SIZES } from './button_icon';

describe('OuiButtonIcon', () => {
  test('is rendered', () => {
    const { container } = render(
      <OuiButtonIcon iconType="user" {...requiredProps} />
    );

    expect(container).toMatchSnapshot();
  });

  describe('props', () => {
    describe('isDisabled', () => {
      it('is rendered', () => {
        const { container } = render(
          <OuiButtonIcon iconType="user" aria-label="button" isDisabled />
        );

        expect(container).toMatchSnapshot();
      });

      it('renders a button even when href is defined', () => {
        const { container } = render(
          <OuiButtonIcon
            iconType="user"
            aria-label="button"
            href="#"
            isDisabled
          />
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('iconType', () => {
      it('is rendered', () => {
        const { container } = render(
          <OuiButtonIcon aria-label="button" iconType="user" />
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('color', () => {
      COLORS.forEach((color) => {
        test(`${color} is rendered`, () => {
          const { container } = render(
            <OuiButtonIcon iconType="user" aria-label="button" color={color} />
          );

          expect(container).toMatchSnapshot();
        });
      });
    });

    describe('display', () => {
      DISPLAYS.forEach((display) => {
        test(`${display} is rendered`, () => {
          const { container } = render(
            <OuiButtonIcon
              iconType="user"
              aria-label="button"
              display={display}
            />
          );

          expect(container).toMatchSnapshot();
        });
      });
    });

    describe('size', () => {
      SIZES.forEach((size) => {
        test(`${size} is rendered`, () => {
          const { container } = render(
            <OuiButtonIcon iconType="user" aria-label="button" size={size} />
          );

          expect(container).toMatchSnapshot();
        });
      });
    });

    describe('isSelected', () => {
      it('is rendered as true', () => {
        const { container } = render(
          <OuiButtonIcon iconType="user" aria-label="button" isSelected />
        );

        expect(container).toMatchSnapshot();
      });

      it('is rendered as false', () => {
        const { container } = render(
          <OuiButtonIcon
            iconType="user"
            aria-label="button"
            isSelected={false}
          />
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('href', () => {
      it('secures the rel attribute when the target is _blank', () => {
        const { container } = render(
          <OuiButtonIcon
            iconType="user"
            aria-label="button"
            href="#"
            target="_blank"
          />
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('onClick', () => {
      it('supports onClick and href', async () => {
        const handler = jest.fn();
        render(
          <OuiButtonIcon
            iconType="user"
            aria-label="hoi"
            href="#"
            onClick={handler}
          />
        );

        const user = userEvent.setup();
        await act(async () => {
          await user.click(screen.getByRole('link'));
        });

        expect(handler).toHaveBeenCalledTimes(1);
      });

      it('supports onClick as a button', async () => {
        const handler = jest.fn();
        render(
          <OuiButtonIcon iconType="user" aria-label="hoi" onClick={handler} />
        );

        const user = userEvent.setup();
        await act(async () => {
          await user.click(screen.getByRole('button'));
        });

        expect(handler).toHaveBeenCalledTimes(1);
      });
    });
  });
});
