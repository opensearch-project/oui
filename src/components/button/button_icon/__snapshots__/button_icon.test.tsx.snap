// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiButtonIcon is rendered 1`] = `
<div>
  <button
    aria-label="aria-label"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall testClass1 testClass2"
    data-test-subj="test subject string"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props color accent is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--accent ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props color danger is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--danger ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props color ghost is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--ghost ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props color primary is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props color subdued is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--subdued ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props color success is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--success ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props color text is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--text ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props color warning is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--warning ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props display base is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props display empty is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props display fill is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--fill ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props href secures the rel attribute when the target is _blank 1`] = `
<div>
  <a
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall"
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </a>
</div>
`;

exports[`OuiButtonIcon props iconType is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props isDisabled is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon-isDisabled ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall"
    disabled=""
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props isDisabled renders a button even when href is defined 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon-isDisabled ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall"
    disabled=""
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props isSelected is rendered as false 1`] = `
<div>
  <button
    aria-label="button"
    aria-pressed="false"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props isSelected is rendered as true 1`] = `
<div>
  <button
    aria-label="button"
    aria-pressed="true"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props size m is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--medium"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props size s is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--small"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;

exports[`OuiButtonIcon props size xs is rendered 1`] = `
<div>
  <button
    aria-label="button"
    class="ouiButtonIcon ouiButtonIcon--primary ouiButtonIcon--empty ouiButtonIcon--xSmall"
    type="button"
  >
    <span
      aria-hidden="true"
      class="ouiButtonIcon__icon"
      color="inherit"
      data-ouiicon-type="user"
    />
  </button>
</div>
`;
